spec/classes/config/global/auth_spec.rb

```diff
@@ -68,6 +78,72 @@
       expect(content).to include('testuser:password,user,deploy')
       expect(content).to include('anotheruser:anotherpassword,user')
     end
+
+    it 'should contain PropertyFileLoginModule and be required' do
+      jaas_auth = catalogue.resource('file', '/etc/rundeck/jaas-auth.conf')[:content]
+      expect(jaas_auth).to include('org.eclipse.jetty.plus.jaas.spi.PropertyFileLoginModule required')
+    end
+  end
+
+  describe 'with multiauth ldap and file auth users array' do
+    let(:params) do
+      {
+        :auth_types  => %w(ldap file),
+        :auth_config => {
+          'file' => {
+            'auth_users' => [
+              {
+                'username' => 'testuser',
+                'password' => 'password',
+                'roles'    => %w(user deploy),
+              },
+              {
+                'username' => 'anotheruser',
+                'password' => 'anotherpassword',
+                'roles'    => ['user'],
+              },
+            ],
+          },
+
+          'ldap' => {
+            'debug'                   => 'true',
+            'url'                     => 'localhost:389',
+            'force_binding'           => 'true',
+            'force_binding_use_root'  => 'true',
+            'bind_dn'                 => 'test_rundeck',
+            'bind_password'           => 'abc123',
+            'user_base_dn'            => 'ou=users,ou=accounts,ou=corp,dc=xyz,dc=com',
+            'user_rdn_attribute'      => 'sAMAccountName',
+            'user_id_attribute'       => 'sAMAccountName',
+            'user_password_attribute' => 'unicodePwd',
+            'user_object_class'       => 'user',
+            'role_base_dn'            => 'ou=role based,ou=security,ou=groups,ou=test,dc=xyz,dc=com',
+            'role_name_attribute'     => 'cn',
+            'role_member_attribute'   => 'member',
+            'role_object_class'       => 'group',
+            'supplemental_roles'      => 'user',
+            'nested_groups'           => 'true',
+          },
+        },
+      }
+    end
+
+    it 'should generate valid content for realm.properties' do
+      content = catalogue.resource('file', '/etc/rundeck/realm.properties')[:content]
+      expect(content).to include('admin:admin,user,admin,architect,deploy,build')
+      expect(content).to include('testuser:password,user,deploy')
+      expect(content).to include('anotheruser:anotherpassword,user')
+    end
+
+    it 'should contain PropertyFileLoginModule and be required' do
+      jaas_auth = catalogue.resource('file', '/etc/rundeck/jaas-auth.conf')[:content]
+      expect(jaas_auth).to include('org.eclipse.jetty.plus.jaas.spi.PropertyFileLoginModule required')
+    end
+
+    it 'should contain JettyCachingLdapLoginModule and be required' do
```

Comment:
This is an example of multiauth, where ldap authentication is used in addition to file based auth, so we're looking for both the strings 'PropertyFileLoginModule' and 'JettyCachingLdapLoginModule'.  The resulting jaas-auth.conf will look something like this:

```
authentication {

  com.dtolabs.rundeck.jetty.jaas.JettyCachingLdapLoginModule sufficient
  debug="true"
  contextFactory="com.sun.jndi.ldap.LdapCtxFactory"
  providerUrl="localhost:389"
  authenticationMethod="simple"
  forceBindingLogin="true"
  forceBindingLoginUseRootContextForRoles="true"
  bindDn="test_rundeck"
  bindPassword="abc123"
  userBaseDn="ou=users,ou=accounts,ou=corp,dc=xyz,dc=ad"
  userRdnAttribute="sAMAccountName"
  userIdAttribute="sAMAccountName"
  userPasswordAttribute="unicodePwd"
  userObjectClass="user"
  roleBaseDn="ou=role based,ou=security,ou=groups,ou=test,dc=xyz,dc=com"
  roleNameAttribute="cn"
  roleMemberAttribute="member"
  roleObjectClass="group"
  supplementalRoles="user"
  cacheDurationMillis="300000"
  reportStatistics="true"
  nestedGroups="true";


  org.eclipse.jetty.plus.jaas.spi.PropertyFileLoginModule required
  debug="true"
  file="/etc/rundeck/realm.properties";

};
```